src/StatusQ/Controls/StatusBaseInput.qml

```diff
@@ -127,191 +110,130 @@ Item {
+                    Layout.fillWidth: true
+                    Layout.topMargin: root.topPadding
+                    Layout.bottomMargin: root.bottomPadding
 
-                    Keys.onEnterPressed: {
-                        if (multiline) {
-                            event.accepted = false
-                        } else {
-                            event.accepted = true
-                        }
+                    contentWidth: edit.paintedWidth
+                    contentHeight: edit.paintedHeight
+                    boundsBehavior: Flickable.StopAtBounds
+                    QC.ScrollBar.vertical: QC.ScrollBar {
+                        interactive: multiline
+                        enabled: multiline
                     }
+                    clip: true
+
+                    function ensureVisible(r) {
+                        if (contentX >= r.x)
+                            contentX = r.x
+                        else if (contentX + width <= r.x + r.width)
+                            contentX = r.x + r.width - width
+                        if (contentY >= r.y)
+                            contentY = r.y
+                        else if (contentY + height <= r.y + r.height)
+                            contentY = r.y + r.height - height
+                    }
+
+                    TextEdit {
+                        id: edit
+
+                        property string previousText: text
+                        property var keyEvent
 
-                    Keys.forwardTo: [statusBaseInput]
-
-                    onTextChanged: {
-                        statusBaseInput.dirty = true
-                        if (statusBaseInput.maximumLength > 0) {
-                            if (text.length > statusBaseInput.maximumLength) {
-                                var cursor = cursorPosition;
-                                text = previousText;
-                                if (cursor > text.length) {
-                                    cursorPosition = text.length;
-                                } else {
-                                    cursorPosition = cursor-1;
+                        width: flick.width
+                        height: flick.height
+                        verticalAlignment: Text.AlignVCenter
+                        selectByMouse: true
+                        selectionColor: Theme.palette.primaryColor2
+                        selectedTextColor: color
+                        focus: true
+                        font.pixelSize: 15
+                        font.family: Theme.palette.baseFont.name
+                        color: Theme.palette.directColor1
+                        wrapMode: root.multiline ? Text.WrapAtWordBoundaryOrAnywhere : TextEdit.NoWrap
+
+                        Keys.onReturnPressed: event.accepted = !multiline
+                        Keys.onEnterPressed: event.accepted = !multiline
+                        Keys.forwardTo: [statusBaseInput]
```

Comment:
This looks like an assumption. Even if the control is designed to be embedded in StatusInput not referencing instances outside this component would make it clear.

Also, this generates a QML warning used on its own for which I propose the following test to catch it (note that it is based on this PR): https://github.com/status-im/StatusQ/pull/590
```suggestion
                        Keys.forwardTo: [root.input]
```

Reply:
Great catch!

Reply:
@osmaczko thanks for pointing out that the correct solution is
```suggestion
                        Keys.forwardTo: [root]
```